libsql-sqlite3/src/insert.c

```diff
@@ -2692,9 +2692,32 @@ int sqlite3OpenTableAndIndices(
       p5 = 0;
     }
     if( aToOpen==0 || aToOpen[i+1] ){
+      /*
+       * sqlite3OpenTableAndIndices is called for 'PRAGMA integrity_check' and
+       * we can't emit OP_OpenVectorIdx command for this operation;
+       *
+       * As vector index creates empty B-tree index - it's safe to issue
+       * OP_OpenRead command for it
+       *
+       * TODO: with current implementation, integrity_check will output error
+       * for vector index as rows will be missed in it
+       * It's better to remove this error in future - but for now it's unclear
+       * how to do that with minimal code changes
+       */
```

Comment:
does that mean that we can't perform integrity checks if vectors are used?

Reply:
You can, but they will report integrity violation warnings for vector indices (e.g. `{row 1 missing from index t_integrity_idx} {wrong # of entries in index t_integrity_idx}`)

Do we use them anywhere in the platform for some needs?

Reply:
We have used them in the past to restore corrupted database